src/EventSubscriber/AddContentLanguageHeaderSubscriber.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\HttpCache\EventSubscriber;
+
+use eZ\Publish\API\Repository\Values\Content\Content;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+final class AddContentLanguageHeaderSubscriber implements EventSubscriberInterface
+{
+    public const CONTENT_LANGUAGE_HEADER = 'x-lang';
+    private const CONTENT_VIEW_ROUTES = ['_ez_content_view', 'ez_urlalias'];
+
+    /** @var bool */
+    private $isTranslationAware;
+
+    public function __construct(bool $isTranslationAware)
+    {
+        $this->isTranslationAware = $isTranslationAware;
+    }
+
+    public function onKernelResponse(ResponseEvent $event)
+    {
+        if (!$this->isTranslationAware || HttpKernelInterface::MASTER_REQUEST != $event->getRequestType()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $content = $request->attributes->get('content');
+        if (
+            $content instanceof Content &&
+            in_array($request->get('_route'), self::CONTENT_VIEW_ROUTES)
+        ) {
```

Comment:
Can we check if view is instance of `\Ibexa\Core\MVC\Symfony\View\ContentValueView` and `\Ibexa\Core\MVC\Symfony\View\CachableView` ? 

See 
https://github.com/ibexa/http-cache/blob/main/src/lib/EventSubscriber/HttpCacheResponseSubscriber.php#L44-L47 for inspiration.